File: rb/features.rb
# StarTrek SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module StarTrekFeatures
  def self.make_feature(name)
    case name
    when "base"
      StarTrekBaseFeature.new
    when "ratelimit"
      StarTrekRatelimitFeature.new
    when "retry"
      StarTrekRetryFeature.new
    when "test"
      StarTrekTestFeature.new
    when "timeout"
      StarTrekTimeoutFeature.new
    else
      StarTrekBaseFeature.new
    end
  end
end

File: py/features.py
# StarTrek SDK feature factory

from feature.base_feature import StarTrekBaseFeature
from feature.test_feature import StarTrekTestFeature


def _make_feature(name):
    features = {
        "base": lambda: StarTrekBaseFeature(),
        "test": lambda: StarTrekTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
